R/utilities-data-combined.R

```diff
@@ -0,0 +1,164 @@
+# TODO: move these functions to the utilities package
+
+#' Count number of objects
+#'
+#' @details
+#'
+#' If the argument is not a vector, unlike `length()`, this function will not
+#' count the number of named bindings in an environment, but only the number of
+#' instances of a class.
+#'
+#' For example, `length(mtcars)` will return 11, but `objCount(mtcars)` will
+#' return 1.
+#'
+#' @param x An object (an atomic vector, a list, or instance(s) of a class).
+#'
+#' @examples
+#'
+#' objCount(c(1, 2, 3)) # 3
+#' objCount(list("a", "b")) # 2
+#' objCount(mtcars) # 1
+#'
+#' @return Integer representing the count of objects.
+#'
+#' @keywords internal
+#' @noRd
+
+objCount <- function(x) {
+  # `is.vector()` can handle both atomic vectors and lists, i.e.
+  # both `is.vector(c(1, 2))` and `is.vector(list(1, 2))` will be `TRUE`
+  if (is.vector(x)) {
+    l <- length(x)
+  } else {
+    l <- length(list(x))
+  }
+
+  return(l)
+}
+
+#'  Validate arguments provided as vectors
+#'
+#' @details
+#'
+#' Validation of arguments provided as a vector involves:
+#'
+#' - Checking that it is of expected length.
+#' - Checking for `NULL` or other unexpected values (`NaN`, `Inf`, `NA` of the
+#'   wrong type) and standardizing them to `NA` of desired type.
+#' - Checking that each element in the vector is of expected type.
+#' - If a non-atomic list is provided, converting it to an atomic vector.
+#'
+#' @param x A vector of arguments.
+#' @param expectedLength An integer to denote the expected length of the vector.
+#' @inheritParams flattenList
+#'
+#' @return
+#'
+#' An atomic vector of desired type containing specified arguments.
+#'
+#' @examples
+#'
+#' validateVectorArgs(list(1, 2, NA, NULL), 4L, "numeric")
+#' validateVectorArgs(c(1, 2, NA, NA_complex), 4L, "numeric")
+#'
+#' @keywords internal
+#' @noRd
+
+validateVectorArgs <- function(arg = NULL, expectedLength = NULL, type) {
```

Comment:
Naming might be a bit misleading because the function actually returns a modified values of args potentially whereas all other validate functions have no return values... I am not good at naming things lol but I would not have expected a returned value (also the examples are not showing this)

Reply:
Hmm, good point.

How about `cleanVectorArgs()`?